### PR TITLE
5589 Fix memory leaks in TSK

### PR DESCRIPTION
**Pull request by jkho**

### CODE REVIEW

File: tsk/base/mymalloc.c [hunk 1]
```diff
@@ -44,10 +44,15 @@ tsk_malloc(size_t len)
 void *
 tsk_realloc(void *ptr, size_t len)
 {
-    if ((ptr = realloc(ptr, len)) == 0) {
```
Comment (bcarrier) [inline tsk/base/mymalloc.c]: I don't see what these changes fix?  Aren't they just the same thing, but more verbose?

Comment (jkho) [reply tsk/base/mymalloc.c]: If/when realloc failed, ptr is assigned to NULL. The original ptr is lost so a memory leak.
This was reported by VLD.

Comment (jkho) [reply tsk/base/mymalloc.c]: https://stackoverflow.com/questions/4548385/realloc-leaking-memory

Comment (bcarrier) [reply tsk/base/mymalloc.c]: Oh, we should make a comment then to that effect.  Otherwise, someone will return it back to what it was.

Comment (jkho) [reply tsk/base/mymalloc.c]: done

**Pull request merged**